discord/guild.py

```diff
@@ -1885,20 +1897,54 @@ async def edit(
 
             fields["features"] = features
 
-        if premium_progress_bar_enabled is not MISSING:
-            fields["premium_progress_bar_enabled"] = premium_progress_bar_enabled
-
         if disable_invites is not MISSING:
-            features = self.features.copy()
+            features = fields.get("features", self.features.copy())
             if disable_invites:
-                if not "INVITES_DISABLED" in features:
+                if "INVITES_DISABLED" not in features:
                     features.append("INVITES_DISABLED")
             else:
                 if "INVITES_DISABLED" in features:
                     features.remove("INVITES_DISABLED")
 
             fields["features"] = features
 
+        if discoverable is not MISSING:
+            features = fields.get("features", self.features.copy())
+            if discoverable:
+                if "DISCOVERABLE" not in features:
+                    features.append("DISCOVERABLE")
+            else:
+                if "DISCOVERABLE" in features:
+                    features.remove("DISCOVERABLE")
+
+            fields["features"] = features
+
+        if raid_alerts is not MISSING:
+            features = fields.get("features", self.features.copy())
+            if raid_alerts:
+                if "RAID_ALERTS_DISABLED" in features:
+                    features.remove("RAID_ALERTS_DISABLED")
+            else:
+                if "RAID_ALERTS_DISABLED" not in features:
+                    features.append("RAID_ALERTS_DISABLED")
+
+            fields["features"] = features
+
+        if enable_activity_feed is not MISSING:
+            features = fields.get("features", self.features.copy())
+            if enable_activity_feed:
+                if "ACTIVITY_FEED_ENABLED_BY_USER" not in features:
+                    features.append("ACTIVITY_FEED_ENABLED_BY_USER")
+                if "ACTIVITY_FEED_DISABLED_BY_USER" in features:
+                    features.remove("ACTIVITY_FEED_DISABLED_BY_USER")
+            else:
+                if "ACTIVITY_FEED_ENABLED_BY_USER" in features:
+                    features.remove("ACTIVITY_FEED_ENABLED_BY_USER")
+                if "ACTIVITY_FEED_DISABLED_BY_USER" not in features:
+                    features.append("ACTIVITY_FEED_DISABLED_BY_USER")
+
+            fields["features"] = features
+
```

Comment:
Instead of computing `features = fields.get("features", self.features.copy())` again and again in every block, we can just do it once at the start of the checks for guild features and after the checks compare the result with `self.features` and set `fields["features"]` if necessary.

Reply:
Good point, will commit asap